api/build.gradle

```diff
@@ -1,32 +1,74 @@
 description = "reactive-streams"
-def jdkFlow = false
+def isJdk9OrGreater = false
 try {
     Class.forName("java.util.concurrent.Flow")
-    jdkFlow = true
+    isJdk9OrGreater = true
 } catch (ClassNotFoundException cnfe) {
 
 }
 
 sourceSets {
     main {
         java {
-            if (jdkFlow)
-              srcDirs = ['src/main/java', 'src/main/java9']
-            else
-              srcDirs = ['src/main/java']
+            srcDirs = ['src/main/java']
         }
     }
+    if (isJdk9OrGreater) java9 {
+        java {
+            srcDirs = ['src/main/java', 'src/main/java9']
+        }
+    }
+}
+
+configurations {
+  mrjarArtifact
 }
 
 jar {
+    if (isJdk9OrGreater) into('META-INF/versions/9') {
+        from(sourceSets.java9.output) {
+            include '**/FlowAdapters*'
+            include '**/module-info.class'
+        }
+    }
+
     bnd ('Bundle-Name': 'reactive-streams-jvm',
             'Bundle-Vendor': 'Reactive Streams SIG',
             'Bundle-Description': 'Reactive Streams API',
             'Bundle-DocURL': 'http://reactive-streams.org',
             'Bundle-Version': project.version,
-            'Export-Package': 'org.reactivestreams.*',
-            'Automatic-Module-Name': 'org.reactivestreams',
-            'Bundle-SymbolicName': 'org.reactivestreams.reactive-streams'
+            'Export-Package': 'org.reactivestreams.*,!META-INF.*',
+            'Multi-Release': true,
+            'Bundle-SymbolicName': 'org.reactivestreams.reactive-streams',
+            '-fixupmessages': '^Classes found in the wrong directory: .*'
```

Comment:
`Export-Package` has added `!META-INF.*` to avoid processing the Java 9 class root; `Multi-Release` was added, to replace `Automatic-Module-Name`. `-fixupmessages` avoids a BND warning about classes in the Java 9 root.